extensions/kubectl-cli/src/extension.ts

```diff
@@ -337,7 +337,44 @@ async function postActivate(
   let releaseToInstall: KubectlGithubReleaseArtifactMetadata | undefined;
   let releaseVersionToInstall: string | undefined;
   let currentVersion = kubectl.version;
-
+  // check if there is a new version to be installed and register the updater
+  let releaseToUpdateTo: KubectlGithubReleaseArtifactMetadata | undefined;
+  let releaseVersionToUpdateTo: string | undefined;
+  const latestAsset = await kubectlDownload.getLatestVersionAsset();
+  const update = {
+    version: latestAsset.tag.slice(1) !== kubectlCliTool.version ? latestAsset.tag.slice(1) : undefined,
+    selectVersion: async (): Promise<string> => {
+      const selected = await kubectlDownload.promptUserForVersion(currentVersion);
+      releaseToUpdateTo = selected;
+      releaseVersionToUpdateTo = selected.tag.slice(1);
+      return releaseVersionToUpdateTo;
+    },
+    doUpdate: async (): Promise<void> => {
+      if (!releaseToUpdateTo || !releaseVersionToUpdateTo) {
+        if (latestAsset) {
+          releaseToUpdateTo = latestAsset;
+          releaseVersionToUpdateTo = latestAsset.tag.slice(1);
+        } else {
+          throw new Error(`Cannot update ${path}. No release selected.`);
+        }
+      }
+      // download, install system wide and update cli version
+      const binaryPath = await kubectlDownload.download(releaseToUpdateTo);
+      await installBinaryToSystem(binaryPath, 'kubectl');
+      kubectlCliTool?.updateVersion({
+        version: releaseVersionToUpdateTo,
+        installationSource: 'extension',
+      });
+      currentVersion = releaseVersionToUpdateTo;
+      if (releaseToUpdateTo === latestAsset) {
```

Comment:
If latest release installed `version` property is removed to switch card to show `Upgrade/Downgrade` command.